runners/core-java/src/main/java/org/apache/beam/runners/core/metrics/MetricsContainerImpl.java

```diff
@@ -383,4 +409,70 @@ public boolean equals(@Nullable Object object) {
   public int hashCode() {
     return Objects.hash(stepName, counters, distributions, gauges);
   }
+
+  private boolean equalsMetricName(MetricName metricName, String namespace, String name) {
+    return (namespace == null || namespace.equals(metricName.getNamespace()))
```

Comment:
This equality looks werid, why would it be considered equal if the param is null?

Can you add a clairifying comment

Reply:
Yeah, `equals` prefix is misleading. It's more like matching and when the field will be ignored when the parameter is null. I renamed the method.